PopoverMenu.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useRef, useState, useEffect } from 'react';
+// This is an SVG of three dots - it has gone through preprocessing and will render when used as <EllipsesMenuIcon />
+import EllipsesMenuIcon from './EllipsesMenuIcon.svg';
+
+interface PopoverMenuItem {
+  name: string;
+  onSelect: () => void;
+  key: number;
+  href?: string;
+}
+
+interface PopoverMenuProps {
+  menuItems: PopoverMenuItem[];
+  useButtonForSecondItem: boolean;
+}
+
+function FocusableLinks({
+  menuItems,
+  closeMenu,
+  useButtonForSecondItem = false,
+}: PopoverMenuProps & { closeMenu: () => void }) {
+  const linkRefs = useRef<(HTMLAnchorElement | HTMLButtonElement | null)[]>([]);
+
+  useEffect(() => {
+    linkRefs.current[0]?.focus();
+  }, [linkRefs]);
+
+  /**
+   * Handles the blur event for the menu item
+   * @param index - index of the menu item
+   */
+  const handleBlur = (index: number) => {
+    if (index === menuItems.length - 1) {
+      closeMenu();
+    }
+  };
+
+  return (
+    <ul>
+      {menuItems.map((item, index) => (
+        <React.Fragment key={item.key}>
+          <li>
+            {useButtonForSecondItem && index === 1 ? (
```

Comment:
to make sure this could be used other then just having the second button, lets remove it and instead tweak the code

`{useButtonForSecondItem && index === 1` to be  `{item.type === "button" (we added item.type in after line 9 in the interface  PopoverMenuItem